aws-api/src/main/java/com/amplifyframework/api/aws/SubscriptionOperation.java

```diff
@@ -88,11 +88,8 @@ public synchronized void start() {
                 },
                 onNextItem,
                 apiException -> {
-                    // Guard against calling something that's been cancelled already.
-                    if (!canceled.get()) {
-                        canceled.set(true);
-                        onSubscriptionError.accept(apiException);
-                    }
+                    cancel();
```

Comment:
This was a bug.  We can't just set `canceled` to true.  We have to actually do the cancellation too.

I've also removed the `!canceled.get()` guard, because the `cancel` method itself is already `idempotent`, and once a subscription is cancelled, it won't throw any additional errors.